src/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -160,6 +160,17 @@ public static byte[] ConvertAlpnProtocolListToByteArray(List<SslApplicationProto
                     done = Interop.OpenSsl.DoSslHandshake(((SafeDeleteSslContext)context).SslContext, inputBuffer.token, inputBuffer.offset, inputBuffer.size, out output, out outputSize);
                 }
 
+                // When the handshake is done, and the context is server, check if the alpnHandle was freed during ALPN.
+                // If it was freed, then that indiciates ALPN failed, send failure.
+                // We have this workaround, as openssl supports terminating handshake only from version 1.1.0,
+                // whereas ALPN is supported from version 1.0.2.
+                if (done && sslAuthenticationOptions.IsServer &&
+                    sslAuthenticationOptions.ApplicationProtocols != null && ((SafeDeleteSslContext)context).SslContext.AlpnHandle.IsAllocated &&
+                    ((SafeDeleteSslContext)context).SslContext.AlpnHandle.Target == null)
```

Comment:
Minor, but it'd be better not to cast multiple times and rather just get the AlpnHandle once and then check IsAllocated and Target.

Reply:
@Maoni0, are there any concerns with setting GCHandle.Target to null (and then freeing the handle later)?  Presumably that has the effect of removing the pin on the previous target, but still maintaining the slot associated with the GCHandle until it's later freed?